### PR TITLE
CLOUDP-67623: Update golangci-lint

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-67623

Closes #[issue number]

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

Update golangci-lint and add some more linters

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -57,16 +61,15 @@ linters:
     - whitespace
 
   # don't enable:
-  # - gochecknoinits
-  # - funlen
-  # - lll
-  # - gomnd
   # - asciicheck
   # - gochecknoglobals
```
Comment (savannaendicott) [inline .golangci.yml:65]: nit: for alphabetical order please put this package after dupl and funlen

File: internal/store/store.go [hunk 2]
```diff
@@ -31,8 +32,16 @@ import (
...
 const (
-	atlasAPIPath = "api/atlas/v1.0/"
-	yes          = "yes"
+	atlasAPIPath          = "api/atlas/v1.0/"
+	yes                   = "yes"
+	responseHeaderTimeout = 10 * time.Minute
+	tlsHandshakeTimeout   = 10 * time.Second
+	timeout               = 10 * time.Second
+	keepAlive             = 30 * time.Second
+	maxIdleConns          = 10
```
Comment (savannaendicott) [inline internal/store/store.go:41]: Is this meant to be `100`? If not, why the drastic change?

Comment (gssbzn) [reply internal/store/store.go:41]: originally I got this from some recommendations for web server, but the cli should not use more than 3 or 4 connections given it's "one off" nature

File: .golangci.yml (hunk 1 above)
Comment (gssbzn) [reply .golangci.yml:65]: 🤦  I thought I sorted'em

**Pull request merged**